test/e2e/volume_test.go

```diff
@@ -0,0 +1,321 @@
+package e2e_test
+
+import (
+	"fmt"
+
+	"github.com/appscode/go/crypto/rand"
+	"github.com/appscode/stash/apis"
+	"github.com/appscode/stash/apis/stash/v1beta1"
+	"github.com/appscode/stash/pkg/util"
+	"github.com/appscode/stash/test/e2e/framework"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+)
+
+var (
+	bpv              *core.PersistentVolume
+	bpvc             *core.PersistentVolumeClaim
+	rpvc             *core.PersistentVolumeClaim
+	pod              core.Pod
+	updateStatusFunc v1beta1.Function
+	backupFunc       v1beta1.Function
+	restoreFunc      v1beta1.Function
+	backupTask       v1beta1.Task
+	restoreTask      v1beta1.Task
+)
+
+var _ = Describe("Volume", func() {
+	BeforeEach(func() {
+		f = root.Invoke()
+
+	})
+	JustBeforeEach(func() {
+		pod = f.Pod(bpvc.Name)
+		cred = f.SecretForLocalBackend()
+		if missing, _ := BeZero().Match(cred); missing {
+			Skip("Missing repository credential")
+		}
+		pvc = f.GetPersistentVolumeClaim()
+		err = f.CreatePersistentVolumeClaim(pvc)
+		Expect(err).NotTo(HaveOccurred())
+		repo = f.Repository(cred.Name, pvc.Name)
+		fmt.Println("pvc for repo", pvc.Name)
+		targetref = v1beta1.TargetRef{}
+		backupCfg = f.BackupConfiguration(repo.Name, targetref)
+		backupCfg.Spec.Target = &v1beta1.Target{
```

Comment:
Move this part to a function and then just use:
```
backupCfg.Spec.Target = framework.PvcBackupTarget(pvc)
```

You can directly use the function as:
```
backupCfg = f.BackupConfiguration(repo.Name, framework.PvcBackupTarget(pvc))
```